www_src/pages/login/form-input.jsx

```diff
@@ -3,13 +3,21 @@ var React = require('react');
 var FormInput = React.createClass({
   getDefaultProps: function () {
     return {
-      type: 'text'
+      type: 'text',
+      tabIndex: 0
     };
   },
+  checkForReturn: function (e) {
+    if (e.keyCode === 13) {
+      if (typeof this.props.onReturn === 'function') {
```

Comment:
`undefined` and `false` are fine. If that's all they can be, no need to be overly safe. If we do explicitly the property, with "not a function", then this code is better.